index.html

```diff
@@ -3957,77 +4078,73 @@ <h3>Algorithm</h3>
           of <var>value</var>.</li>
-        <li>Otherwise, if <var>value</var> has an <code>@language</code> <a>entry</a> whose
-          value matches the <a>language mapping</a> of
-          <var>active property</var>, return the value associated with the
-          <code>@value</code> <a>entry</a> of <var>value</var>.</li>
-        <li>Otherwise, if <var>number entries</var> equals <code>1</code> and either
-          the value of the <code>@value</code> <a>entry</a> is not a <a>string</a>,
-          or the <var>active context</var> has no <a>default language</a>,
-          or the <a>language mapping</a> of <var>active property</var>
-          is set to <code>null</code>, return the value associated with the
-          <code>@value</code> <a>entry</a>.</li>
-        <li>Otherwise:
-          <ol class="changed">
-            <li>Replace any value of <code>@type</code> in <var>value</var> with the result of using the
+        <li class="changed">Otherwise, if the <a>type mapping</a> of <var>active property</var> is <code>@none</code>,
+          or <var>value</var> has an `@type` <a>entry</a>,
+          and the value of `@type` in <var>value</var> does not match the <a>type mapping</a> of <var>active property</var>,
+          leave <var>value</var> as is, as value compaction is disabled.
+          <ol>
+            <li>Replace any value of <code>@type</code> in <var>result</var> with the result of using the
               <a href="#iri-compaction">IRI compaction algorithm</a>,
               passing <var>active context</var>, <var>inverse context</var>,
               the value of the <code>@type</code> <a>entry</a> for <var>var</var>, and
               <code>true</code> for <var>vocab</var>.</li>
-            <li>Replace each key in <var>value</var> with the result of using the
-              <a href="#iri-compaction">IRI compaction algorithm</a>,
-              passing <var>active context</var>, <var>inverse context</var>,
-              the that key for <var>var</var>, and
-              <code>true</code> for <var>vocab</var>.</li>
-            <li>Return <var>value</var>.</li>
           </ol>
         </li>
+        <li>Otherwise, if the value of the <code>@value</code> <a>entry</a> is not a <a>string</a>:
+          <ol>
+            <li>If <var>value</var> has an `@index` <a>entry</a>,
+              and the <a>container mapping</a> associated to <var>active property</var>
+              includes <code>@index</code>,
+              set <var>result</var> to the value associated with the `@value` <a>entry</a>.</li>
+          </ol>
+        </li>
+        <li>If the value of `@language` in <var>item</var> exactly matches <var>language</var>,
+          if it is not `null`, or is not present, if it is `null`,
+          <span class="changed">and the value of `@direction` in <var>item</var> exactly matches <var>direction</var>,
+            if it is not `null`, or is not present, if it is `null`</span>:
```

Comment:
I think this would be clearer
```suggestion
            if it is not `null`, or is not present, if <var>direction</var> is `null`</span>:
```